backend/doc/migration.md

```diff
@@ -34,7 +34,397 @@ and activate the Tracim virtualenv:
+set the type.
+
+```python
+from alembic import op
+def upgrade():
+    with op.batch_alter_table("workspaces") as bop:
+        bop.alter_column("calendar_enabled", new_column_name="agenda_enabled")
+```
+
+will work on both postgresql and sqlite but will failed on mysql, you need to specify
+type_ like this :
+
+
+```python
+from alembic import op
+from sqlalchemy import Boolean
+
+def upgrade():
+    with op.batch_alter_table("workspaces") as bop:
+        bop.alter_column("calendar_enabled", new_column_name="agenda_enabled", type_=Boolean)
+    # ### end Alembic commands ###
+
+
+def downgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table("workspaces") as bop:
+        bop.alter_column("agenda_enabled", new_column_name="calendar_enabled", type_=Boolean)
+```
+
+see [32e629b17e2e_change_calendar_enabled_to_agenda_enabled_colum.py]("../tracim_backend/migration/32e629b17e2e_change_calendar_enabled_to_agenda_enabled_colum.py")
+
+This example is a bit tricky because, it is in fact, not working properly on recent mariadb and mysql (8+ version),
+this is related to issue in alembic support for these database.
+
+A fix that should work on any database, should be :
+
+```python
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy import Boolean
+
+def rename_boolean_column(
+    table_name: str, old_column_name: str, new_column_name: str, constraint_name: str
+):
+    """
+    Utils to help rename a boolean column supporting:
+    - postgresql
+    - mysql 8.0+
+    - mariadb 10.3+
+    - sqlite
+    """
+    with op.batch_alter_table(table_name) as bop:
+        dialect = op.get_context().dialect
+        if dialect.name == "mysql":
+            if not dialect._is_mariadb and dialect.server_version_info >= (8, 0, 0):
+                bop.drop_constraint(constraint_name, type_="check")
+            bop.alter_column(
+                old_column_name,
+                new_column_name=new_column_name,
+                type_=Boolean(create_constraint=False),
+            )
+            bop.alter_column(new_column_name, type_=Boolean(constraint_name))
+        else:
+            bop.alter_column(old_column_name, new_column_name=new_column_name)
+
+def upgrade():
+    rename_boolean_column("workspaces", "calendar_enabled", "agenda_enabled", "workspaces_chk_2")
+
+
+def downgrade():
+    rename_boolean_column("workspaces", "agenda_enabled", "calendar_enabled", "workspaces_chk_2")
+
+```
+
+As you see, for current alembic version of tracim (1.0.5), you need to:
+- drop explictyl constraint for mysql 8.0+ version
+- do rename for all mysql based database (mariadb included) without creating constraint, then reapply type with
+constraint to be sure constraint will be readded.
+
+## Mysql does already add index to foreign keys
+
+Mysql database automatically add index to foreign keys, this lead to issue with migration downgrade as we cannot drop these index.
```

Comment:
adds
this leads

issues 

with migration downgrade -> when downgrading?

these indexes